XenAdmin/TabPages/GeneralTabPage.cs

```diff
@@ -926,8 +926,27 @@ private void GenerateBootBox()
             {
                 s.AddEntry(FriendlyName("VM.BootOrder"), HVMBootOrder(vm),
                     new PropertiesToolStripMenuItem(new VmEditStartupOptionsCommand(Program.MainWindow, vm)));
+
                 if (Helpers.NaplesOrGreater(vm.Connection))
-                    s.AddEntry(FriendlyName("VM.BootMode"), HVMBootMode(vm));
+                {
+                    if (vm.IsHVM() && vm.IsDefaultBootModeUefi())
+                    {
+                        var secureBoot = vm.GetSecureBootMode();
+                        var pool = Helpers.GetPoolOfOne(vm.Connection);
+                        var poolHasCertificates = !string.IsNullOrEmpty(pool?.uefi_certificates);
+
+                        if (secureBoot == "true" && !poolHasCertificates)

```

Comment:
Though it is worth mentioning that technically this is also incorrect, as the value is sometimes "false", as you will have noticed.

> If it is present
and populated then treat it as `true`, otherwise treat it as `false` (and grey out the option for
enabling secureboot). 


However, the specs I am speaking of are linked to the original ticket (called Install uefi-certificates with RPMs)

Reply:
Practically the logic is the way you describe it, however, I wrote it this way on purpose in order to make the specs more explicit, i.e. `auto` is the same as `true` but only if the certificates are there, otherwise it is the same as `false`. The greying out bit is not very correct, please have a look at the discussion on XSI-1362.

Reply:
I added a comment.